server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1986,27 +1962,22 @@ private static ImmutableOpenMap<ShardId, SnapshotsInProgress.ShardSnapshotStatus
                         shardRepoGeneration = null;
                     }
                     final ShardSnapshotStatus shardSnapshotStatus;
-                    if (indexRoutingTable == null) {
```

Comment:
`indexRoutingTable` is accessed a bit further above. If it were null, it would have raised NPEs before hitting this.